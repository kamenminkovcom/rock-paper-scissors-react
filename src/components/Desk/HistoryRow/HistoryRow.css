.history-row {
    height: 20%;
}