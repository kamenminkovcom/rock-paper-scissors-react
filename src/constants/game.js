export const handTypes = {
    paper: 'paper',
    rock: 'rock',
    scissors: 'scissors',
};

export const gameRules = {
    [handTypes.paper]: {
        [handTypes.paper]: 0,
        [handTypes.rock]: 1,
        [handTypes.scissors]: 0,
    },
    [handTypes.rock]: {
        [handTypes.rock]: 0,
        [handTypes.scissors]: 1,
        [handTypes.paper]: 0
    },
    [handTypes.scissors]: {
        [handTypes.scissors]: 0,
        [handTypes.paper]: 1,
        [handTypes.rock]: 0
    }
};

export const possibleResults = {
    win: 'win',
    lose: 'lose',
    draw: 'draw'
};

//may we would like to change classNames and it should not break the logic
export const classNames = {
    [possibleResults.win]: 'win',
    [possibleResults.lose]: 'lose',
    [possibleResults.draw]: 'draw'
};

export const gameHistoryCount = 5;

export const defaultChoices = {
    cpuChoice: undefined,
    playerChoice: undefined
};