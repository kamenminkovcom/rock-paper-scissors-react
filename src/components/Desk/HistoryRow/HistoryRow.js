import React from 'react';
import PropTypes from 'prop-types';
import clssnames from 'classnames';
import {classNames, possibleResults, handTypes} from '../../../constants/game';

import './HistoryRow.css';

const HistoryRow = ({result, cpuChoice, playerChoice}) => (
    <div className="default-row-container history-row">
        <div className={clssnames(cpuChoice,
            'choice-icon',
            'background-no-repeat',
            'background-contain-size',
            'rotate')}/>
            <span className={clssnames('capitalize', classNames[result])}>{result}</span>
        <div className={clssnames(playerChoice,
            'choice-icon',
            'background-no-repeat',
            'background-contain-size')}/>
    </div>
);

HistoryRow.propTypes = {
    result: PropTypes.oneOf(Object.values(possibleResults)).isRequired,
    cpuChoice: PropTypes.oneOf(Object.values(handTypes)).isRequired,
    playerChoice: PropTypes.oneOf(Object.values(handTypes)).isRequired
};

export default HistoryRow;

