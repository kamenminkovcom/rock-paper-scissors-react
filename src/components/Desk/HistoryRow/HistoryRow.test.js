import React from 'react';
import {shallow} from 'enzyme';
import HistoryRow from './HistoryRow';
import {possibleResults, handTypes} from '../../../constants/game';

describe('<HistoryRow/>', () => {
    it('Should render correctly', () => {
        const wrapper = shallow(<HistoryRow result={possibleResults.win}
                                            playerChoice={handTypes.rock}
                                            cpuChoice={handTypes.paper}/>);
        expect(wrapper).toMatchSnapshot();
    })
});