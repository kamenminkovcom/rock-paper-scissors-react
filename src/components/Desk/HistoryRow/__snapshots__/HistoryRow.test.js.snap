// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoryRow/> Should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HistoryRow
    cpuChoice="paper"
    playerChoice="rock"
    result="win"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="paper choice-icon background-no-repeat background-contain-size rotate"
/>,
        <span
          className="capitalize win"
>
          win
</span>,
        <div
          className="rock choice-icon background-no-repeat background-contain-size"
/>,
      ],
      "className": "default-row-container history-row",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "paper choice-icon background-no-repeat background-contain-size rotate",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "win",
          "className": "capitalize win",
        },
        "ref": null,
        "rendered": "win",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "rock choice-icon background-no-repeat background-contain-size",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="paper choice-icon background-no-repeat background-contain-size rotate"
/>,
          <span
            className="capitalize win"
>
            win
</span>,
          <div
            className="rock choice-icon background-no-repeat background-contain-size"
/>,
        ],
        "className": "default-row-container history-row",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "paper choice-icon background-no-repeat background-contain-size rotate",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "win",
            "className": "capitalize win",
          },
          "ref": null,
          "rendered": "win",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "rock choice-icon background-no-repeat background-contain-size",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
